doc_src/cmds/fish_command_not_found.rst

```diff
@@ -0,0 +1,77 @@
+.. _cmd-fish_cmd_not_found:
+
+fish_command_not_found - what to do when a command wasn't found
+===============================================================
+
+Synopsis
+--------
+
+::
+
+  function fish_command_not_found
+      ...
+  end
+
+
+Description
+-----------
+
+When fish can't find a command, it invokes this function.
+
+It can print a message, usually tailored to the OS, to e.g. check which package would contain the missing command,
+by calling out to a handler provided by the operating system.
+
+Fish ships multiple handlers for various Linux distributions and chooses from them when this function is loaded,
+or you can define your own.
+
+It receives the full commandline as one argument per token, so $argv[1] contains the missing command.
```

Comment:
A slightly different, less honest approach for these three paragraphs. Not sure if this is any better:

By default, it prints a message to inform you which package would contain the missing command.
This is done by delegating to an operating-system specific handler. Fish knows about the right handler for various Linux distributions and chooses the appropriate one when this function is loaded.

If you don't want that behavior, you can define `fish_command_not_found` yourself.
It receives the full commandline as one argument per token, so $argv[1] is the missing command.